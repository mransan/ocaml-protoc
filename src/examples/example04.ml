open Example04_types
open Example04_pp

let (@+) value next = 
  Cons {value;next}

let () = 
  let l = 1 @+ 2 @+ 3 @+ 4 @+ 5 @+ Nil in 
  Format.(fprintf std_formatter "l = %a\n" pp_int_list l)
