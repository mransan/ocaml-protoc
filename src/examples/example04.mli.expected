
(** Code for example04.proto *)

(* generated from "example04.proto", do not edit *)



(** {2 Types} *)

type int_list_nil = unit

type int_list_cons = {
  value : int;
  next : int_list;
}

and int_list_t =
  | Cons of int_list_cons
  | Nil

and int_list = {
  t : int_list_t option;
}


(** {2 Basic values} *)

val default_int_list_nil : unit
(** [default_int_list_nil ()] is the default value for type [int_list_nil] *)

val default_int_list_cons : 
  ?value:int ->
  ?next:int_list ->
  unit ->
  int_list_cons
(** [default_int_list_cons ()] is the default value for type [int_list_cons] *)

val default_int_list_t : unit -> int_list_t
(** [default_int_list_t ()] is the default value for type [int_list_t] *)

val default_int_list : 
  ?t:int_list_t option ->
  unit ->
  int_list
(** [default_int_list ()] is the default value for type [int_list] *)


(** {2 Formatters} *)

val pp_int_list_nil : Format.formatter -> int_list_nil -> unit 
(** [pp_int_list_nil v] formats v *)

val pp_int_list_cons : Format.formatter -> int_list_cons -> unit 
(** [pp_int_list_cons v] formats v *)

val pp_int_list_t : Format.formatter -> int_list_t -> unit 
(** [pp_int_list_t v] formats v *)

val pp_int_list : Format.formatter -> int_list -> unit 
(** [pp_int_list v] formats v *)
